### PR TITLE
Feat/make quarter dates amendable in contentful

### PR DESCRIPTION
**Pull request by stringiest**
Adds a new QuarterDates GraphQL query to retrieve quarter dates information from Contentful, allowing Content Designers to update quarter dates themselves rather than requiring developer support.

New Content Type of [Energy table - content pattern](https://app.contentful.com/spaces/vms0u05139aw/environments/master/content_types/contentPatternString/fields) created in Contentful.

Two pieces of content created using this content type
- [Energy CSR quarter dates](https://app.contentful.com/spaces/vms0u05139aw/environments/master/entries/3OdGRuiq5a99kVxYpQQklF?focusedField=contentPatternStringName)
- [Energy CSR next quarter release date](https://app.contentful.com/spaces/vms0u05139aw/environments/master/entries/230w36VUwaerIjibJYFoSG?focusedField=contentPatternStringName)

Initial draft of the GraphQL query matched against the `name` field for pieces of content of that type, as it made the code readable, but the `name` field is not fixed and so may be changed by content designers. Have refactored to use `contentful_id` instead.

N.B. Use of Contentful data currently conditional on feature flag `FF_CONTENTFUL_QTR_DATES` being enabled. This env var is currently not set in `qa` or `prod`, so will need to be set to `true` in `qa` before merging this PR to `main`. This is to allow for testing in a non-prod environment.

### CODE REVIEW

**Comment by stringiest** [comment]
See [slack thread](https://acab.slack.com/archives/CSGEFQ6SD/p1717489742310429) for discussion with Andrew Johnson and @KPobeeNorris about naming of the new content type in Contentful, and how to lock it down to avoid misuse. 

Name agreed to be changed from the initially-proposed 'Content pattern string' to 'Energy table - content pattern'. Name remains unchanged in the background, and is still `contentPatternString`, so no changes to the code have been needed.

File: app/controllers/suppliers_controller.rb [hunk 1]
```diff
@@ -57,6 +59,16 @@ def supplier_with_top_three
     @supplier_with_top_three ||= Supplier.fetch_with_top_three(permitted_params[:id])
   end
 
+  def set_quarter_date
+    # passes in contentful id for quarter dates content
+    @quarter_date = QuarterDate.fetch_quarter_dates_content("3OdGRuiq5a99kVxYpQQklF")
```
Comment (davidsauntson) [inline app/controllers/suppliers_controller.rb:64]: Do you think it's likely these ids will change for any reason?

If you miss the readability of the names, you could use constants eg

```
class SuppliersController < ApplicationController
  QUARTER_DATE_CONTENT_ID = "3OdGRuiq5a99kVxYpQQklF"
  NEXT_QUARTER_DATE_CONTENT_ID = "230w36VUwaerIjibJYFoSG"
 ...
```

File: app/views/suppliers/index.html.haml [hunk 2]
```diff
@@ -9,10 +9,18 @@
...
-    We will publish updated scores for the following 3 months in June 2024.
+  - if Feature.enabled? "FF_CONTENTFUL_QTR_DATES"
+    %h2
+      Scores for #{quarter_date.body}
+
+    %p
+    We will publish updated scores for the following 3 months in #{next_quarter_release.body}.
+  - else
+    %h2
+      Scores for plain string October to December 2023
```
Comment (davidsauntson) [inline app/views/suppliers/index.html.haml:20]: `plain string` should be something else I think

File: app/views/suppliers/index.html.haml [hunk 3]
```diff
@@ -9,10 +9,18 @@
...
+
+    %p
+      We will publish updated scores for the following 3 months in plain string June 2024.
```
Comment (davidsauntson) [inline app/views/suppliers/index.html.haml:23]: `plain string` should be something else I think

**Review by davidsauntson — COMMENTED** [review]
Looks good.

I am worried about hard-coding ids in case they change for some reason, but I understand the trade off between that and potential breaking changes to the `name` in Contentful.

The `plain string` parts need changing.  A belt and braces approach would be to test the component with the feature flag on and off - if we had done that there would have been a failure.  

I think there are some new features coming up that will be using feature flags, so maybe we should introduce `ClimateControl` or something now - depends if you think there is time before this quarter's launch.  It doesn't have to be in this PR.

File: app/views/suppliers/index.html.haml (hunk 2 above)
Comment (stringiest) [reply app/views/suppliers/index.html.haml]: Whoops! Fixed - thanks!

File: app/views/suppliers/index.html.haml (hunk 3 above)
Comment (stringiest) [reply app/views/suppliers/index.html.haml]: Fixed - thanks!

File: app/controllers/suppliers_controller.rb (hunk 1 above)
Comment (stringiest) [reply app/controllers/suppliers_controller.rb]: I hope they won't change, but can't guarantee that no-one would ever delete them.

Good idea re using constants - thanks!

**Review by davidsauntson — APPROVED** [review]

**Pull request merged**